inst/pages/98_exercises.qmd

```diff
@@ -568,42 +568,183 @@ colData(tse)$Barcode_full_length
+
+head(unique(rowData(tse)$Phylum)) # Features
+
+unique(tse$SampleType) # Samples
+```
+
+3.2. Subset the TreeSE with your found samples and features.
+
+```{r}
+#| label: subsetting-ex3.2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+features <- c("Crenarchaeota","Euryarchaeota","Actinobacteria")
+samples <- c("Feces", "Skin", "Tongue")
+
+tse_subset_by_feature <- tse[rowData(tse)$Phylum %in% features
+                             ,
+                             tse$SampleType %in% samples]
+
+unique(rowData(tse_subset_by_feature)$Phylum)
+```   
 
 
 ### Library sizes
 
 1. Calculate library sizes  
 2. Subsample / rarify the counts (see: rarefyAssay)
 
-Useful functions: nrow, ncol, dim, summary, table, quantile, unique, scater::addPerCellQC, mia::mergeFeaturesByRank
+Useful functions: nrow, ncol, dim, summary, table, quantile, unique, scater::addPerCellQC, mia::agglomerateByRank
 
 ### Prevalent and core taxonomic features
 
-1. Estimate prevalence for your chosen feature (row, taxonomic group)  
-2. Identify all prevalent features and subset the data accordingly   
-3. Report the thresholds and the dimensionality of the data before and after subsetting  
-4. Visualize prevalence  
+1. Estimate prevalence for your chosen feature (specific row and taxonomic group)
+```{r}
+#| label: prevalence-ex1
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+# First, merge the features by rank and add it as an alternative experiment.
+altExp(tse,"Phylum") <- agglomerateByRank(tse, "Phylum") 
+
+# Then get the prevalence for a specific sample 
+getPrevalence(altExp(tse,"Phylum"), detection = 1/100, 
+                   sort = TRUE, assay.type = "counts",
+                   as_relative = TRUE)["Phylum:Proteobacteria"]
+```   
+2. Identify all prevalent features and subset the data accordingly.
+
+```{r}
+#| label: prevalence-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+# Get the prevalent features
+prevalentFeatures <- getPrevalentFeatures(tse, detection = 0, 
+                                          prevalence = 50/100)
+# Subset the data to the prevalent features
+prevalenceTse <- tse[rowData(tse)$Phylum %in% prevalentFeatures,]
+
+# Alternatively subsetByPrevalentFeatures can be used to subset directly
+prevalenceTse <- subsetByPrevalentFeatures(tse, detection = 0, 
+                                          prevalence = 50/100)
+```  
+3. How many features are left ?
+
+```{r}
+#| label: prevalence-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+length(assay(prevalenceTse, "counts"))
+```  
+
+4. Recalculate the most prevalent features based on relative abundance. 
+How many are left?
+
+```{r}
+#| label: prevalence-ex4
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
+tse <- transformAssay(tse, MARGIN = "samples", method="relabundance")
+
+prevalenceTse <- subsetByPrevalentFeatures(tse, detection = 0, 
```

Comment:
detection=0 will yield the same results, it might be more informative to use detection = 1 for count assay example, and detection = 0.1/100 for relabundance example to explain  & show the difference?